docs/whats_new_40.rst

```diff
@@ -0,0 +1,135 @@
+What's New in ReFrame 4.0
+=========================
+
+ReFrame 4.0 introduces some important new features and removes all features, configuration options and interfaces that were deprecated in the 3.x versions.
+It also introduces a couple of new deprecations.
+
+ReFrame 4.0 maintains backward compatibility as much as possible.
+Existing 3.x configurations and 3.x tests are expected to run out-of-the-box, despite any warnings issued.
+Framework's behavior with respect to performance logging has also changed, but configuration options are offered so that users can switch to the old behavior.
+
+This page summarizes the key changes in ReFrame 4.0 and what users should pay attention to.
+
+For a complete list of changes, please refer to the `Release Notes <https://github.com/reframe-hpc/reframe/releases/tag/v4.0.0>`__.
+
+
+New Features and Enchancements
+------------------------------
+
+Chaining Configuration Files
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There is no need anymore to keep a huge configuration file with all your system and environment definitions and it is no more required to carry on the generic system configuration as well as any part of the builtin configuration.
+ReFrame 4.0 allows you to split your configuration in multiple files.
+This allows you to create minimal configuration files that contain only the necessary parts.
+For example, if you want to define a general configuration parameter, you don't need to copy the builtin configuration file and add it, but you simply add it in a single ``general`` section.
+This can also be very useful if you maintain a ReFrame installation used by others, as you can update your settings (systems, environments and other options) and any of your users' custom configuration will automatically inherit your settings if it is properly chained.
+To assist with system-wide installation the ``RFM_CONFIG_PATH`` environment variable is introduced that allows you to specify a path where ReFrame will look for configuration files to load.
+
+Now that systems and environments definitions can be distributed over multiple configuration files, it can become easy to accidentally redefine a system or environment without a notice.
+For this reason, ReFrame warns you if a system or an environment are redefined in the same scope.
+Since in the past all configuration files where extended copies of the builtin, you will get warnings that the ``generic`` system and the ``builtin`` environment are redefined, as ReFrame finds them in the builtin configuration, which is always loaded.
```

Comment:
```suggestion
Since in the past all configuration files where extended copies of the ``builtin``, you will get warnings that the ``generic`` system and the ``builtin`` environment are redefined, as ReFrame finds them in the builtin configuration, which is always loaded.
```

Reply:
I mean "builtin configuration" here; I will update the text accordingly.